src/utils/eth_transaction.cairo

```diff
@@ -31,27 +31,36 @@ namespace EthTransaction {
         let (items: RLP.Item*) = alloc();
         RLP.decode(items, tx_data_len, tx_data);
 
-        // the tx is a list of fields, hence first level RLP decoding
-        // is a single item, which is indeed the sought list
         assert [items].is_list = TRUE;
-        let sub_items_len = [items].data_len;
-        let sub_items = cast([items].data, RLP.Item*);
+        let items_len = [items].data_len;
```

Comment:
I don't like shadowing the previous definition of `items`

Reply:
it depends on the context but here the first `items` it not realy the list items, it's the struct containing the items.
I think that it's clearer to read items[0], items[1], etc. when you think about the tx, because it's like in regular languages, where the tx would be `[item, item, item]`. You don't call "item" "sub_item" here.

Reply:
rename first `items` to `payload` maybe? thus we have payload which is a RLP-list, we decode to a cairo-list of items